src/strategies/adaptive.jl

```diff
@@ -0,0 +1,302 @@
+"""
+    AdaptiveParticleSwarm(n_particles = 3,
+                          w = 0.9,
+                          c1 = 2.0,
+                          c2 = 2.0,
+                          prob_shift = 0.25,
+                          rng = Random.GLOBAL_RNG)
+
+Instantiate an adaptive particle swarm optimization tuning strategy. A swarm is
```

Comment:
I suggest that for maintenance reasons we not simply repeat those bits of the doc-string that are the same as the non-adaptive case. Either:

- Simply state what is the same as in the vanilla case (which can be referenced as "\[\`ParticleSwarm\`\]\(\@ref\)")
- Define strings that get interpolated into both doc-strings.